client/src/js/components/dateEditor.js

```diff
@@ -0,0 +1,46 @@
+/**
+* The date editor component
+*
+* @module components/dateEditor
+*
+* @description a component to deal with date, it let you choose a date
+* This component can do his own validation based on parameters values provided*
+* 
+**/ 
+
+/**
+* A date editor component controller
+* @param {object} $scope an angular object which expose the model to the view
+* @param {object} $element an angular object, which present a json form of an element
+* @param {object} $attrs an angular object, which present a json form of attributs
+**/
+
+function dateEditorController ($scope, $element, $attrs, util){
+	var ctrl = this;
+
+	this.$onInit = function (){
```

Comment:
Also, you don't need the $onInit method.  You can write its contents in the main body of this controller without needing to use a special angular method.